### PR TITLE
fix: jsonOpAsync for non object args

### PR DESCRIPTION
**Pull request by lucacasonato**
This fixes two bugs:

1. errors in the synchronous part of jsonOpAsync on the rust side were not propagated to JS correctly
2. only JS objects could be used as arguments. Now primitives like string, number, boolean, or array are also supported

cc @cab @bartlomieju

### CODE REVIEW

File: core/ops.rs [hunk 1]
```diff
@@ -218,16 +219,19 @@ where
   F: Fn(Rc<RefCell<OpState>>, Value, BufVec) -> R + 'static,
   R: Future<Output = Result<Value, AnyError>> + 'static,
 {
+  #[derive(Deserialize)]
+  #[serde(rename_all = "camelCase")]
+  struct Params {
+    promise_id: u64,
+    args: Value,
+  }
```
Comment (bartlomieju) [inline core/ops.rs:227]: IIRC this was removed by @piscisaureus for the performance reasons

Comment (lucacasonato) [reply core/ops.rs:227]: Can you elaborate? We could change this to a JSON array. I think that would have less overhead than the previous impl.

Comment (bartlomieju) [reply core/ops.rs:227]: The idea was that getting only `promise_id` was faster than trying to do deserialization into struct here and then once again in the actual op. Maybe we should guard arguments passed to `jsonOpAsync` (and sync as well) in `core.js`?

Comment (lucacasonato) [reply core/ops.rs:227]: If that is the case, array should be just as fast. Let me implement that.

**Comment by lucacasonato** [comment]
Closing in favor of https://github.com/denoland/deno/pull/9423

**Pull request closed without merging**